approach_mission.md

```diff
@@ -0,0 +1,164 @@
+
+![](88.jpg)
+
+
+
+Continue with our story - 
+
+
+> Registration expired after a year. The user will get a reminder emails - 3 month, 1 month and 1 day before expired, so he can renew his registration.
+ 
+
+What information do we need in order to send the reminder emails?
+
+We need to know of course when the membership began, then we can calculate when 9 months has passed and send the first reminder email. In technical words, the beginning date (or any other date represent something occurs in the system), calls `timestamp`.
+
+But this information is not enough. Think about situation user decide not to renew his registration, what happened to his membership? Is it going away, delete from the system? Well, usually we don’t delete content from the system, but we can mark it as inactive.
+
+So in order to send the user reminder, we need to know two things:  the membership created date (i.e `timestamp`) and if it is active or not (i.e `status`).
+
+![](9.jpg)
+
+Now we have all the information we need for sending the first reminder email (the one that comes after 9 month), so in order to know to which user we need to send that email, we need to retrieve from the database the right user. 
+Let’s describe in words the query for getting the right information from database:
+
+
+Give me all `membership` that their `status` is **active** and their `timestamp` is **today's date minus 9 month**.
+
+
+Ok, we almost there, but there is still one more thing - we need to check that we don't send the email more then one time to the same user. Assuming that we have a lots of `membership` in our database, so we need to limit the number of `membership` we get every time we run the query (if the system will bring us all the fit `membership` at ones, it can be run out of memory). So for example we will tell the system to brings us only 100 `membership` at a time, and we will run the query every 5 min. 
+
+How do we make sure that the system won't bring us the same `membership` we already sent email for 5 min ago?
+
+We need to save the information for which `membership` we already sent emails, and check it every time we want to send reminder email.
+But where are we going to save this information?
+Is it going to be at the `user` entity? Well, remember that `user` can have more than one `membership`, so it will be very complex to put all this data at the `user` entity. Additionally, the email sends based on a membership's timestamp, so it makes more sense that the `membership` will hold the information about the emails who was send. Well, this won't be good enough ether, because we need to send three emails (3 month, 1 month and 1 day before the membership expire), and it's to much data to put at at the `membership` entity.
+
+In that case it will be better to create new entity. Let's call it `email log`.
```

Comment:
@amitaibu please review this new part about email log, I want to make sure the explanation is correct before @RachelBaram do English review.